cogs/birthday/birthday.py

```diff
@@ -519,46 +556,119 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
+                    response = await self.bot.wait_for("message", timeout=30.0, check=check)
+                except asyncio.TimeoutError:
+                    await ctx.send(f"{headerBad}: You took too long. Not setting your birthday.")
+                    return
 
+                if response.content.lower() != "yes":
+                    await ctx.send(f"{headerBad}: Declined. Not setting your birthday.")
+                    return
+                can_delete = ctx.channel.permissions_for(ctx.me).manage_messages
+
+                if can_delete:
+                    msg: discord.Message = await ctx.send(f"{headerWarn}: {confirmationStr}")
+                    try:
+                        response = await self.bot.wait_for("message", timeout=10.0, check=check)
+                    except asyncio.TimeoutError:
+                        try:
+                            await ctx.send(f"{headerBad}: No response. Aborting", delete_after=10)
+                            await msg.delete()
+                            return
+                        except (discord.NotFound, discord.HTTPException):
+                            return
+
+                    if response.content.lower() != "yes":
+                        await ctx.send(
+                            f"{headerBad}: Declined. Not setting your birthday.", delete_after=5
+                        )
+                        try:
+                            await msg.delete()
+                        except (discord.NotFound, discord.HTTPException):
+                            return
+                        return
+
+                    await birthdayConfig.get_attr(KEY_BDAY_MONTH).set(birthday.month)
+                    await birthdayConfig.get_attr(KEY_BDAY_DAY).set(birthday.day)
+                    await birthdayConfig.get_attr(KEY_ADDED_BEFORE).set(True)
+                    await ctx.send(
+                        f"{headerGood}: Successfully set your birthday to "
+                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}.",
+                        delete_after=5,
+                    )
+                    self.logger.info(
+                        "%s#%s (%s) added their birthday as %s",
+                        ctx.author.name,
+                        ctx.author.discriminator,
+                        ctx.author.id,
+                        birthdayStr,
+                    )
+                    await self.checkBirthday()
+                    try:
+                        await msg.delete()
+                        return
+                    except (discord.NotFound, discord.HTTPException):
+                        return
+
+                    else:
+                        await ctx.send(
+                            f"{headerBad}: Unable to delete messages. Please "
+                            "notify the server administrators to give me the "
+                            '"Manage Messages" permission'
```

Comment:
I don't think users need to know about the "Manage Messages" permission. Can just tell them to contact a server admin.
```suggestion
                            f"{headerBad}: Unable to delete messages. Please "
                            "notify a server administrator."
```

Reply:
Removed